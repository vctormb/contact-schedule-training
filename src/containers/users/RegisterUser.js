import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import RegisterUserForm from './RegisterUserForm';

export class RegisterUser extends Component {
    componentDidMount() {
        const { match, dispatch } = this.props;

        // if params has id, it is to edit the user
        if (match.params.id) {
            dispatch({
                type: 'FETCH_USER_REQUEST',
                payload: { userId: match.params.id }
            });
        } else { // if params has no id, reset the user reducer to reset the redux-form
            dispatch({ type: 'REGISTER_USER_RESET' });
        }
    }

    handleSubmit = (values) => {
        const { dispatch } = this.props;

        alert(JSON.stringify(values, null, 2));

        dispatch({
            type: 'ADD_USER_REQUEST',
            push: this.props.history.push
        });
    }

    render() {
        return (
            <div>
                <RegisterUserForm onSubmit={this.handleSubmit} />
                <br />
                <span><Link to="/users">Back to users list</Link></span>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        usersReducer: state.users
    }
}

export default connect(mapStateToProps)(RegisterUser);