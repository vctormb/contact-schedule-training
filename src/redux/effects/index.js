import { all } from 'redux-saga/effects';
import { userSagas } from '../effects/users/sagas';

export default function* rootSagas() {

    // here we initialize all the sagas from different files
    yield all([
        ...userSagas,
    ]);
}