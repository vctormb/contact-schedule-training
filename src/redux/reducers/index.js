import { combineReducers } from 'redux';
import users from './users';

// all the reducers in one place
const rootReducers = combineReducers({
    users: users,
});

export default rootReducers;