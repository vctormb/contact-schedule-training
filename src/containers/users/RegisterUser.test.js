import React from 'react';
import PropTypes from 'prop-types';
import { RegisterUser } from './RegisterUser';
import { shallow, mount } from 'enzyme';

import createRouterContext from 'react-router-test-context';

import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducers from '../../redux/reducers';

let store = createStore(rootReducers);

describe('<RegisterUserForm />', () => {
    const dispatch = jest.fn();

    let props = {
        dispatch,
        match: { params: { id: 0 } },
    };

    const context = createRouterContext();
    RegisterUser.contextTypes = {
        router: PropTypes.object
    }

    let registerUser = shallow(<RegisterUser {...props} />);

    it('should render properly', () => {
        expect(registerUser).toMatchSnapshot();
    });

    // START
    
    describe('when componentDidMount()', () => { // <----------------
        beforeEach(() => {
            registerUser = mount(
                <Provider store={store}>
                    <RegisterUser {...props} />
                </Provider>,
                { context }
            );
        });

        it('should dispatch REGISTER_USER_RESET action', () => { 
            expect(dispatch).toBeCalledWith({ type: 'REGISTER_USER_RESET' });
        });
    });

    // END
});