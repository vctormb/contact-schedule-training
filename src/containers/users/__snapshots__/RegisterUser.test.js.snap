// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterUserForm /> should render properly 1`] = `
<div>
  <withRouter(Connect(ReduxForm))
    onSubmit={[Function]}
  />
  <br />
  <span>
    <Link
      replace={false}
      to="/users"
    >
      Back to users list
    </Link>
  </span>
</div>
`;
